docs/ruleset_functions/compare_schedules.md

```diff
@@ -0,0 +1,46 @@
+
+## compare_schedules
+
+Description: This function would compare two schedules and determine if they match with or without a comparison factor when applicable.  
+
+Inputs:
+- **schedule_1**: First schedule.
+- **schedule_2**: Second schedule.
+- **mask_schedule**: The schedule that defines comparison mode for all 8760 hours in a year, i.e. if hourly value is 0, schedule_1 shall be equal to schedule_2; if hourly value is 1, schedule_1 shall be equal to schedule_2 times the comparison factor; if hourly value is 2, skip comparison for that hour;  
+- **comparison_factor**: The target multiplier number for schedule_1 compared to schedule_2, i.e. when applicable, the hourly value in schedule_1 shall be equal to that in schedule_2 times the comparison_factor.
+
+Returns:
+- **compare_schedules_result_array**: An array containing the total number of hours that the function compares and the number of hours schedule_1 matches schedule_2 with the comparison_factor, i.e. [total_hours_compared, total_hours_match]
+
+Function Call:
+
+- get_schedule_8760_values_array()
+
+Data Lookup: None
+
+Logic:
+
+- Get 8760 hourly values array for schedule_1, schedule_2 and mask_schedule: `hourly_value_sch_1 = get_schedule_8760_values_array(schedule_1), hourly_value_sch_2 = get_schedule_8760_values_array(schedule_2), hourly_value_mask_sch = get_schedule_8760_values_array(mask_schedule)`
+
```

Comment:
@xingcx not sure I understand : is the mask schedule pre defined for each type of comparison ?

Reply:
Right the way the function is written right now we will need to define it in the RDS, might need another function for this.